### ISSUE TITLE
Misleading error message reported for Texting.SendMessageDirect

### ISSUE THREAD

**Issue by espertus** [mit-cml/appinventor-sources#1965]
**Describe the bug**

A misleading error message is displayed when `Texting.SendMessageDirect` is used.

**Affects**

- [ ] Designer
- [X] Blocks editor
- [ ] Companion
- [X] Compiled apps
- [X] Buildserver
- [ ] Other... (please describe)

**Expected behavior**

A better error message would be shown either when adding the block, when building the app, or running the app. Specifically, the user should be directed to information explaining that a different version of the companion needs to be used.

**Steps to reproduce**

1. Install the MIT AI2 companion from the Google Play Store.
2. Create and run an app that uses the Texting.sendMessageDirect block.
3. An error message is shown: "Error 908: The permission SEND_SMS has been denied. Please enable it in the Settings app"

**See Also**

[MIT App Inventor Forum: Error 908: The permission SEND_SMS has been denied. Please enable it](https://groups.google.com/forum/#!msg/mitappinventortest/EyhhXsDy_ZY/rKFlWnNMAQAJ)

**Comment by ewpatton** [comment]
I think we should show a popup the first time the block is used in a project, and we can update the companion code so that if the companion doesn't have the permission (i.e., it's the Google Play companion), then we can report a more useful error. Compiled apps will automatically get the necessary permission and prompt when needed.

**Comment by Abhik1998** [comment]
So should we redirect the user based upon the user's choice in the pop-up showing two options:- deny and allow

**Comment by ewpatton** [comment]
@Abhik1998 I think what we will want to do is have an error message on the companion that is specific to this error that we throw when the conditions are met. Then in the blocks editor we can detect the message and show an internationalized prompt that allows them to install the companion by either opening the Help > Companion Information dialog or by just showing the QR code. The latter requires fewer steps but is a bit more work.

**Comment by ewpatton** [comment]
@Abhik1998 Also, if you are planning to work on this, since it requires changes to the companion you will need to branch off of `ucr` rather than `master`.

**Comment by asayushg** [comment]
Hi, @ewpatton I want to work on this issue if no one is working.

**Comment by ewpatton** [comment]
@Abhik1998 Have you made any progress on this or can @asayushg take it over?

**Comment by Abhik1998** [comment]
Yep, I have almost done it. I will make the PR sometimes. Working upon the QR code part.

**Comment by arinmodi** [comment]
@ewpatton, Is it solved ? Or can i work on this?
